scripts/component_readme/readme_template.md

```diff
@@ -1,31 +1,56 @@
 # {{ name }}
 
-### Description
+## Description
 {{ description }}
 
-### Inputs / outputs
+## Inputs / outputs
 
+### Consumes
 {% if consumes %}
 **This component consumes:**
-
 {% for field_name, field in consumes.items() %}
 - {{ field.name }}: {{ field.type.value }}
 {% endfor %}
-{% else %}
-**This component consumes no data.**
 {% endif %}
 
+{% if is_consumes_generic %}
+**This component can consume additional fields**
+- <field_name>: <dataset_field_name>
+This defines a mapping to update the fields consumed by the operation as defined in the component spec.
+The keys are the names of the fields to be received by the component, while the values are 
+the name of the field to map from the input dataset
+
+See the usage example below on how to define a field name for additional fields.
+
+{% endif %}
+
+{% if not is_consumes_generic and not consumes%}
+**This component does not consume data.**
+{% endif %}
+
+
+
+### Produces
 {% if produces %}
 **This component produces:**
-
```

Comment:
This newline is needed for formatting the list correctly.

Reply:
fixed